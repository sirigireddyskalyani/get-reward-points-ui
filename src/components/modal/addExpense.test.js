import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { screen, render, fireEvent } from '@testing-library/react';

import AddExpense from './addExpense';

const mockStore = configureStore([]);
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useDispatch: () => (jest.fn())
}));
describe('HorizontalNavigator Component Snapshot', () => {
    let component;

    beforeEach(() => {
        const store = mockStore({
            expenses: {
                data: []
            }
        });
        component = renderer.create(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
    });

    it('AddExpense component rendered successfully', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});
describe('Home Container Behavior Testing', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            expenses: {
                data: []
            }
        });
    });

    it('check addExpense form Validations', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        const submitBtn = screen.getAllByTestId('add-expense-submit-btn')[0];
        submitBtn.click();
        expect(container.querySelector('input[name="username"]').value).toEqual("");

        const username = container.querySelector('input[name="username"]');
        await fireEvent.change(username, {
            target: {
                value: "mock@email.com"
            }
        })
        submitBtn.click();
        expect(container.querySelector('input[name="username"]').value).toEqual("mock@email.com");

        const expenses = container.querySelector('input[name="expenses"]');
        await fireEvent.change(expenses, {
            target: {
                value: 0
            }
        })
        submitBtn.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("0");
    });

    it('check addExpense form Validations', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        const username = container.querySelector('input[name="username"]');
        await fireEvent.change(username, {
            target: {
                value: "mock@email.com"
            }
        })
        const expenses = container.querySelector('input[name="expenses"]');
        await fireEvent.change(expenses, {
            target: {
                value: 0
            }
        })

        const submitBtn = screen.getAllByTestId('add-expense-submit-btn')[0];
        submitBtn.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("0");
    });

    it('handle closeModal button click', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        const closeModal = screen.getAllByTestId('add-expense-modal-close-btn')[0];
        closeModal.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("");
    });

    it('handle addExpense button click', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        const addExpenseBtn = screen.getAllByTestId('add-expense-btn')[0];
        addExpenseBtn.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("");
    });


    it('handle add new expense', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        await fireEvent.change(username, { target: { value: "Smith" } });
        await fireEvent.change(expenses, { target: { value: "90" } });

        const submitBtn = screen.getAllByTestId('add-expense-submit-btn')[0];
        submitBtn.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("90");
    });

    it('handle add new expense', async () => {
        const {container} = render(
            <Provider store={store}>
                <AddExpense />
            </Provider>
        );
        await fireEvent.change(username, { target: { value: "Smith" } });

        const submitBtn = screen.getAllByTestId('add-expense-submit-btn')[0];
        submitBtn.click();
        expect(container.querySelector('input[name="expenses"]').value).toEqual("");
    });
});