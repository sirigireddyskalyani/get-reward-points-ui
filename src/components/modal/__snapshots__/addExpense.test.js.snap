// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalNavigator Component Snapshot AddExpense component rendered successfully 1`] = `
Array [
  <label
    className="btn btn-sm btn-info modal-button"
    data-testid="add-expense-btn"
    onClick={[Function]}
  >
    Add Expense
  </label>,
  <div
    className="modal modal-bottom  sm:modal-middle"
  >
    <div
      className="modal-box"
    >
      <div
        className="relative"
      >
        <div
          className="flex justify-between"
        >
          <h2
            className="font-bold text-lg"
          >
            Add Expense
          </h2>
          <label
            className="btn btn-sm btn-circle absolute right-2 top-2"
            data-testid="add-expense-modal-close-btn"
            onClick={[Function]}
          >
            âœ•
          </label>
        </div>
        <div
          className="form-control w-full mt-2"
        >
          <label
            className="label"
          >
            <span
              className="label-text"
            >
              Name
            </span>
            <span
              className="label-text-alt text-red-600"
            >
              *
            </span>
          </label>
          <input
            className="input input-bordered w-full"
            data-testid="add-expense-username-input"
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name ex..customer name"
            type="text"
          />
        </div>
        <div
          className="form-control w-full mt-2"
        >
          <label
            className="label"
          >
            <span
              className="label-text"
            >
              Description
            </span>
          </label>
          <textarea
            className="textarea textarea-bordered"
            id="description"
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="About expense details"
          />
        </div>
        <div
          className="form-control w-full mt-2"
        >
          <label
            className="label"
          >
            <span
              className="label-text"
            >
              Amount
            </span>
            <span
              className="label-text-alt text-red-600"
            >
              *
            </span>
          </label>
          <input
            className="input input-bordered w-full"
            id="expenses"
            name="expenses"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Amount"
            type="number"
          />
        </div>
        <div
          className="flex justify-end m-2"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="btn btn-sm btn-success m-2"
            data-testid="add-expense-submit-btn"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
